contracts/defi/provider/MakerProvider.sol

```diff
@@ -0,0 +1,817 @@
+        removeCollateral(_wallet, _loanId, _collateralAmount, IMakerCdp(_oracles[0]));
+    }
+
+    /**
+     * @dev Increases the debt by borrowing more token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _debtToken The token borrowed.
+     * @param _debtAmount The amount of token to borrow.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     */
+    function addDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        IMakerCdp makerCdp = IMakerCdp(_oracles[0]);
+        require(_debtToken == makerCdp.sai(), "Maker: debt token must be DAI");
+        addDebt(_wallet, _loanId, _debtAmount, IMakerCdp(_oracles[0]));
+    }
+
+    /**
+     * @dev Decreases the debt by repaying some token from a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _debtToken The token to repay.
+     * @param _debtAmount The amount of token to repay.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     */
+    function removeDebt(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address _debtToken, 
+        uint256 _debtAmount, 
+        address[] calldata _oracles
+    ) 
+        external
+    {
+        IMakerCdp makerCdp = IMakerCdp(_oracles[0]);
+        require(_debtToken == makerCdp.sai(), "Maker: debt token must be DAI");
+        removeDebt(_wallet, _loanId, _debtAmount, makerCdp, UniswapFactory(_oracles[1]));
+    }
+
+    /**
+     * @dev Gets information about a loan identified by its ID.
+     * @param _wallet The target wallet.
+     * @param _loanId The ID of the loan if any, 0 otherwise.
+     * @param _oracles (optional) The address of one or more oracles contracts that may be used by the provider to query information on-chain.
+     * @return a status [0: no loan, 1: loan is safe, 2: loan is unsafe and can be liquidated, 3: unable to provide info] and the estimated ETH value of the loan
+     * combining all collaterals and all debts. When status = 1 it represents the value that could still be borrowed, while with status = 2
+     * it represents the value of collateral that should be added to avoid liquidation.      
+     */
+    function getLoan(
+        BaseWallet _wallet, 
+        bytes32 _loanId, 
+        address[] calldata _oracles
+    ) 
+        external 
+        view 
+        returns (uint8 _status, uint256 _ethValue)
+    {
+        IMakerCdp makerCdp = IMakerCdp(_oracles[0]);
+        if(exists(_loanId, makerCdp)) {
+            return (3,0);
+        }
+        return (0,0);
```

Comment:
I don't understand this function. From what I understand, if the CDP has some collateral, we return "no info", otherwise, we return "no loan", why?

Reply:
We cannot wrap the desired methods as the `getLoan` method is `view` and the target methods modify the state. The only info we can provide is if the loan (aka the CDP) exists. If yes we return `status=3` (exists but no info), if no  we return `status=0` (no loan). I've updated the comment to make that clearer.

Reply:
Ok